crates/bevy_reflect/bevy_reflect_derive/src/registration.rs

```diff
@@ -22,9 +23,21 @@ pub(crate) fn impl_get_type_registration(
         }
     });
 
+    // Add Typed bound for each active field
+    let mut where_reflect_typed_clause = if where_clause.is_some() {
+        quote! {#where_clause}
+    } else if !field_types.is_empty() {
+        quote! {where}
+    } else {
+        quote! {}
+    };
+    where_reflect_typed_clause.extend(quote! {
+        #(#field_types: #bevy_reflect_path::Reflect + #bevy_reflect_path::Typed,)*
+    });
```

Comment:
Since this is used throughout multiple modules now, could we add it to `bevy_reflect_derive/src/utility.rs`?

Alternatively, we could add an even more general-purpose version of this to `bevy_macro_utils`.

Reply:
Good idea!